src/main/java/org/kohsuke/github/GHApp.java

```diff
@@ -39,7 +39,9 @@ public GHUser getOwner() {
      *
      * @param owner
      *            the owner
+     * @deprecated Do not use this method. It was added due to incomplete understanding of Jackson binding.
```

Comment:
Looks like jackson code is handled here:

https://github.com/github-api/github-api/blob/796c644c4ac0f4637b6289ed10a1ac8654c7d7b4/src/main/java/org/kohsuke/github/GitHubClient.java#L738

I assume it's doing field injection in someway?

I'm used to either constructor or setter binding with jackson, haven't seen this setup before

Reply:
@timja It's not exactly handled there.  Yes, the creation of the reader and adding of injected values happens there. 

Other than the value injection, binding happens just like it usually does.   
 
The point of this change is that the setter methods for Jackson properties do not need to be public. Jackson can access internal or private setters, and will even automatically bind fields that do not have setters or constructor paramters (regardless of access limitations).  See https://github.com/github-api/github-api/blob/master/src/main/java/org/kohsuke/github/example/dataobject/ReadOnlyObjects.java for some examples of what I've learned.

Reply:
@timja  I'm happy to continue discussing this topic, but it didn't seem like something that needed to block merging this change.